hlx_statics/scripts/scripts.js

```diff
@@ -457,9 +457,53 @@ function loadTitle() {
   document.title = window.location.href;
 }
 
+function loadPrism(document) {
+  const codeBlocks = document.querySelectorAll(
+    'code[class*="language-"], [class*="language-"] code'
+  );
+
+  if (!codeBlocks.length) return; 
+
+  const observer = new IntersectionObserver(
+    (entries, observer) => {
+      entries.forEach((entry) => {
+        if (entry.isIntersecting) {
+          const pre = entry.target.closest('pre') || entry.target;
+          pre.classList.add('prism-loading');
+
+          observer.unobserve(entry.target); 
+          loadPrismScripts(pre);
```

Comment:
So this loading the prism scripts every time it scrolls into view. If you look at the network tab, you can see it makes a lot of requests when scrolling down the test page: https://devsite-1574--adp-devsite--adobedocs.aem.page/express/add-ons/docs/guides/tutorials/grids-addon. Ideally, it should load once it comes into view and not have to load it again.